File: live_streaming_with_cohosting/ZegoUIKitPrebuiltLiveStreamingWithCoHostOC/ZegoUIKitPrebuiltLiveStreamingWithCoHostOC/ViewController.m
//
//  ViewController.m
//  ZegoUIKitPrebuiltLiveStreamingWithCoHostOC
//
//  Created by zego on 2024/5/28.
//

#import "ViewController.h"

@import ZegoUIKit;
@import ZegoUIKitPrebuiltLiveStreaming;
@import ZegoUIKitSignalingPlugin;


@interface ViewController ()<ZegoUIKitPrebuiltLiveStreamingVCDelegate>
@property (nonatomic, copy) NSString *userName;
@property (nonatomic, copy) NSString *userID;
@property (nonatomic, copy) NSString *appSign;
@property (nonatomic, assign) unsigned int appID;

@property (weak, nonatomic) IBOutlet UITextField *roomIDTextField;
@property (weak, nonatomic) IBOutlet UISwitch *aspectFillSwitch;

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  self.userID = [NSString stringWithFormat:@"%u",arc4random() % 99999];
  self.userName = [NSString stringWithFormat:@"name_%@",self.userID];
  self.appID = <#APPID#>;
  self.appSign = @"<#APPSIGN#>";
  self.roomIDTextField.text = [NSString stringWithFormat:@"%u",arc4random() % 99999];
}

- (IBAction)startLive:(id)sender {
  ZegoUIKitPrebuiltLiveStreamingConfig *config = [ZegoUIKitPrebuiltLiveStreamingConfig hostWithEnableSignalingPlugin:YES];
  
  ZegoPrebuiltAudioVideoViewConfig * audioVideoViewConfig = [[ZegoPrebuiltAudioVideoViewConfig alloc] init];
  audioVideoViewConfig.useVideoViewAspectFill = self.aspectFillSwitch.on;
  config.audioVideoViewConfig = audioVideoViewConfig;
  config.enableCoHosting = YES;
  ZegoUIKitPrebuiltLiveStreamingVC * vc = [[ZegoUIKitPrebuiltLiveStreamingVC alloc] init:self.appID appSign:self.appSign userID:self.userID userName:self.userName liveID:self.roomIDTextField.text config:config];
  vc.delegate = self;
  vc.modalPresentationStyle = UIModalPresentationFullScreen;
  [self presentViewController:vc animated:YES completion:^{
    
  }];
}

- (IBAction)watchLive:(id)sender {
  
  ZegoUIKitPrebuiltLiveStreamingConfig *config = [ZegoUIKitPrebuiltLiveStreamingConfig audienceWithEnableSignalingPlugin:YES];
  ZegoPrebuiltAudioVideoViewConfig * audioVideoViewConfig = [[ZegoPrebuiltAudioVideoViewConfig alloc] init];
  audioVideoViewConfig.useVideoViewAspectFill = self.aspectFillSwitch.on;
  config.audioVideoViewConfig = audioVideoViewConfig;
  config.enableCoHosting = YES;
  ZegoUIKitPrebuiltLiveStreamingVC * vc = [[ZegoUIKitPrebuiltLiveStreamingVC alloc] init:self.appID appSign:self.appSign userID:self.userID userName:self.userName liveID:self.roomIDTextField.text config:config];
  vc.delegate = self;
  vc.modalPresentationStyle = UIModalPresentationFullScreen;
  [self presentViewController:vc animated:YES completion:^{
    
  }];
}

#pragma mark - ZegoUIKitPrebuiltLiveStreamingVCDelegate

@end

File: live_streaming/ZegoUIKitPrebuiltLiveStreaming/ZegoUIKitPrebuiltLiveStreamingDemo/ViewController.swift
//
//  ViewController.swift
//  ZegoUIKitPrebuiltLiveStreaming
//
//  Created by zego on 2022/8/22.
//

import UIKit
import ZegoUIKit
import ZegoUIKitPrebuiltLiveStreaming

class ViewController: UIViewController {
    
    let appID: UInt32 = <#YourAppID#>
    let appSign: String = <#YourAppSign#>
    
    @IBOutlet weak var roomIDTextField: UITextField! {
        didSet {
            let roomID: UInt32 = arc4random() % 999999
            roomIDTextField.text = String(format: "%d", roomID)
        }
    }
    
    let userID: String = String(format: "%d", arc4random() % 999999)
    var userName: String?
    

    override func viewDidLoad() {
        super.viewDidLoad()
        self.userName = String(format: "n_%@", self.userID)
        // Do any additional setup after loading the view.
    }
    
    @IBAction func startLive(_ sender: Any) {
        let config: ZegoUIKitPrebuiltLiveStreamingConfig = ZegoUIKitPrebuiltLiveStreamingConfig.host()
        let liveVC: ZegoUIKitPrebuiltLiveStreamingVC = ZegoUIKitPrebuiltLiveStreamingVC(self.appID, appSign: self.appSign, userID: self.userID, userName: self.userName ?? "", liveID: self.roomIDTextField.text ?? "", config: config)
        liveVC.modalPresentationStyle = .fullScreen
        self.present(liveVC, animated: true, completion: nil)
    }
    
    @IBAction func watchLive(_ sender: Any) {
        let config: ZegoUIKitPrebuiltLiveStreamingConfig = ZegoUIKitPrebuiltLiveStreamingConfig.audience()
        let liveVC: ZegoUIKitPrebuiltLiveStreamingVC = ZegoUIKitPrebuiltLiveStreamingVC(self.appID, appSign: self.appSign, userID: self.userID, userName: self.userName ?? "", liveID: self.roomIDTextField.text ?? "", config: config)
        liveVC.modalPresentationStyle = .fullScreen
        self.present(liveVC, animated: true, completion: nil)
    }

}

